lib/utils.py

```diff
@@ -25,31 +25,29 @@ def build(command: List[str], directory: str):
         return False
     return True
 
+def build_architecture(arch, cross, num_cpus, target):
+    print(f"Building {arch}")
+    data = ["make", f"ARCH={arch}", cross, "-j", str(num_cpus), "defconfig", target]
+    subprocess.check_output(data)
+    print(f"{arch} works")
+
 def build_check(target: Path) -> bool:
     '''Checks if the linux kernel builds properly'''
 
     num_cpus = len(os.sched_getaffinity(0))
     shutil.copy('.config', '.tmp.config')
     try:
-        print("Building arm")
-        data = ["make", "ARCH=arm", "LLVM=1", "-j", str(num_cpus), "defconfig", target]
-        subprocess.check_output(data)
-        print("arm works")
-
-        print("Building arm64")
-        data[1] = "ARCH=arm64"
-        subprocess.check_output(data)
-        print("arm64 works")
-
-        print("Building riscv")
-        data[1] = "ARCH=riscv"
-        subprocess.check_output(data)
-        print("riscv works")
-
-        print("Building x86")
-        data[1] = "ARCH=x86"
-        subprocess.check_output(data)
-        print("x86 works")
+        clang_archs = ["arm", "arm64", "mips", "riscv", "powerpc", "x86", "um"]
+        for arch in clang_archs:
+            build_architecture(arch, 'LLVM=1', num_cpus, target)
+
+        architectures = ['alpha', 'arc', 'csky', 'hppa', 'hppa64', 'i386', 'loongarch64', 'm68k', 'microblaze',
+                         'mips64', 'nios2', 'or1k', 's390', 'sh2', 'sh4', 'sparc', 'sparc64', 'xtensa']
+
+        for arch in architectures:
+            if shutil.which(f'{arch}-linux-gcc') is None:
+                continue
+            build_architecture(arch, f'CROSS_COMPILE={arch}-linux-', num_cpus, target)
```

Comment:
consider adding a link to https://mirrors.edge.kernel.org/pub/tools/crosstool/ in the comments